common/funding_tx.c

```diff
@@ -53,3 +56,271 @@ struct bitcoin_tx *funding_tx(const tal_t *ctx,
 	assert(bitcoin_tx_check(tx));
 	return tx;
 }
+
+#if EXPERIMENTAL_FEATURES
+/* We leave out the change addresses if there's no change left after fees */
+static size_t calculate_input_weights(struct input_info **inputs,
+				      struct amount_sat *total)
+{
+	u32 input_weight;
+	u64 scriptlen;
+	size_t weight = 0, i = 0;
+
+	*total = AMOUNT_SAT(0);
+	for (i = 0; i < tal_count(inputs); i++) {
+		/* prev_out hash + index + sequence */
+		input_weight = (32 + 4 + 4) * 4;
+
+		if (inputs[i]->script) {
+			scriptlen = tal_bytelen(inputs[i]->script);
+			input_weight += (scriptlen + varint_size(scriptlen)) * 4;
+		} else {
+			/* 00 byte script_sig len */
+			input_weight += 1 * 4;
+		}
+
+		input_weight += inputs[i]->max_witness_len;
+		weight += input_weight;
+
+		assert(amount_sat_add(total, *total, inputs[i]->input_satoshis));
```

Comment:
Not assert(): see previous "no code inside assert()" advice.

But what if they were to get a tx through to us claiming BIGNUM satoshis incoming?  You need to be explicit and return bool here, all the way through the callchain.